erpnext/stock/doctype/item/item.py

```diff
@@ -33,6 +33,11 @@ def autoname(self):
 		if frappe.db.get_default("item_naming_by")=="Naming Series" and not self.variant_of:
 			from frappe.model.naming import make_autoname
 			self.item_code = make_autoname(self.naming_series+'.#####')
+		elif frappe.db.get_default("item_naming_by")=="Naming Series" and self.variant_of:
```

Comment:
maybe add this in the earlier if so that it does not query twice:
```
if frappe.db.get_default("item_naming_by")=="Naming Series":
  if self.variant_of:
    ...
  else:
    ...
```